tkVideoPlayer/tkvideoplayer.py

```diff
@@ -157,53 +174,127 @@ def _load(self, path):
+                                    audio_device.write(interleaved_data)
+                                    
+                                if self._stop or self._paused:
+                                    break
+                                    
+                        else:
+                            now = time.time_ns() // 1_000_000  # time in milliseconds
+                            delta = now - then  # time difference between current frame and previous frame
+                            then = now
+                             
+                            frame = next(self._container.decode(video=0))
+
+                            self._time_stamp = float(frame.pts * stream.time_base)
+
+                            width = self._current_frame_size[0]
+                            height = self._current_frame_size[1]
+                            if self._keep_aspect_ratio:
+                                im_ratio = frame.width / frame.height
+                                dest_ratio = width / height
+                                if im_ratio != dest_ratio:
+                                    if im_ratio > dest_ratio:
+                                        new_height = round(frame.height / frame.width * width)
+                                        height = new_height
+                                    else:
+                                        new_width = round(frame.width / frame.height * height)
+                                        width = new_width
+
+                            self._current_img = frame.to_image(width=width, height=height, interpolation="FAST_BILINEAR")
+
+                            self._frame_number += 1
+                    
+                            self.event_generate("<<FrameGenerated>>")
 
-                        if self._frame_number % self._video_info["framerate"] == 0:
-                            self.event_generate("<<SecondChanged>>")
+                            if self._frame_number % self._video_info["framerate"] == 0:
+                                self.event_generate("<<SecondChanged>>")
 
-                        if self.consistant_frame_rate:
-                            time.sleep(max((time_in_frame - delta)/1000, 0))
+                            if self.consistant_frame_rate:
+                                time.sleep(max((time_in_frame - delta)/1000, 0))
 
-                        # time.sleep(abs((1 / self._video_info["framerate"]) - (delta / 1000)))
+                            # time.sleep(abs((1 / self._video_info["framerate"]) - (delta / 1000)))
 
                     except (StopIteration, av.error.EOFError, tk.TclError):
                         break
-                    
-                self._container.close()
 
             # print("Container: ", self._container.c)
             if self._container:
                 self._container.close()
+                stream.close()
```

Comment:
Fixes a memory leak

Reply:
Ok, I actually had tried that when upgrading to PyAv 11, but the problem of memory leak started when I was trying to loop the video. Every time it looped the memory starts to rise. Probably a problem with GC, but I can't seem to get it right